core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/base/ODurableComponent.java

```diff
@@ -65,19 +64,22 @@
   protected final OPerformanceStatisticManager performanceStatisticManager;
 
   private volatile String name;
+  private volatile String fileName;
   private volatile String fullName;
 
   protected final String extension;
 
   private volatile String lockName;
 
-  public ODurableComponent(OAbstractPaginatedStorage storage, String name, String extension, String lockName) {
+  public ODurableComponent(OAbstractPaginatedStorage storage, String name, String fileName, String extension, String lockName) {
     super(true);
 
     assert name != null;
+    assert fileName != null;
     this.extension = extension;
     this.storage = storage;
-    this.fullName = name + extension;
+    this.fileName = fileName;
+    this.fullName = fileName + extension;
```

Comment:
That is incorrect you mix full component name and full file name. For example, in start operation, you should use getFullName but when you add a file you should use gitFullFileName.